### PR TITLE
ref(tests): Fix flaky mute digest test

### PR DESCRIPTION
**Pull request by ceorourke**
Protect against 🥐 - sometimes the first message was sent to `self.user` and the 2nd one was sent to `user2`, but sometimes it was the other way around.

### CODE REVIEW

File: tests/sentry/mail/test_adapter.py [hunk 1]
```diff
@@ -1312,13 +1312,21 @@ def test_notify_digest_snooze_one_rule(self, notify):
...
+        else:
+            # user2 receives a digest about both alerts, since no rules were muted
+            assert "2 new alerts since" in message2.subject
+
+        if message1.to[0] == user2.email:
+            # user2 receives a digest about both alerts, since no rules were muted
+            assert "2 new alerts since" in message1.subject
+        else:
+            # self.user only receives a digest about one alert, since a rule was muted
+            assert "1 new alert since" in message1.subject
```
Comment (ceorourke) [inline tests/sentry/mail/test_adapter.py:1330]: idk if there's a better way to do this - I first tried creating both users and a new project and whatnot in the test so it was more contained but that was still flaky.

Comment (ceorourke) [reply tests/sentry/mail/test_adapter.py:1330]: Actually did I just make a test that can never fail ...

Comment (wedamija) [reply tests/sentry/mail/test_adapter.py:1330]: I'd probably do something like
`messages = sorted(mail.outbox, key=lambda message: message.to[0])`  (probably not exactly right code, but the general idea)

Then you have a stable ordering and can rely on them being in the same order every time

**Review by asottile-sentry — APPROVED** [review]

**Review by wedamija — APPROVED** [review]

**Pull request merged**